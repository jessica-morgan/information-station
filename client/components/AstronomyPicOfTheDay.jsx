import { getApod } from '../api/nasaApi'
import React, { useState, useEffect } from 'react'

export const AstronomyPicOfTheDay = () => {
  const [apod, setApod] = useState()
  useEffect(() => {
    getApod()
      .then(apodInfo => {
        setApod({
          title: apodInfo.title,
          image: apodInfo.url,
          description: apodInfo.explanation,
          date: apodInfo.date
        })
      })
  }, [])
  return apod ? (
    <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
      <h1>{apod.title}</h1>
      <img style={{ width: '80%', height: '80%' }} src={apod.image}></img>
      <h3>{apod.date}</h3>
      <h2>{apod.description}</h2>
    </div>) : <div style={{ display: 'block', width: '70vw', height: '100%', float: 'right', position: 'relative' }}>
  loading component
  </div>
}

export default AstronomyPicOfTheDay
