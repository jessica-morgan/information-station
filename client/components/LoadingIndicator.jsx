import React from 'react'

const LoadingIndicator = () => {
  return (
    <div>
      <img src="/loading.gif"></img>
    </div>
  )
}

export default LoadingIndicator
