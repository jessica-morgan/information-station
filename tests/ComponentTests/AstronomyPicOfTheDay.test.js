import React from 'reactn'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import AstronomyPicOfTheDay from '../../client/components/AstronomyPicOfTheDay'
import { mount, shallow } from 'enzyme'
import LoadingIndicator from '../../client/components/LoadingIndicator'
const nasaApi = require('../../client/api/nasaApi')

test('AstronomyPicOfTheDay renders correctly', () => {
  const component = renderer.create(
    <MemoryRouter>
      <AstronomyPicOfTheDay />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('AstronomyPicOfTheDay should render LoadingIndicator if posts is not yet set with api data', done => {
  const wrapper = mount(<AstronomyPicOfTheDay />)
  setImmediate(() => {
    wrapper.update()
    expect(wrapper.containsMatchingElement(<LoadingIndicator></LoadingIndicator>)).toBeTruthy()
    done()
  })
})

test('AstronomyPicOfTheDay contains correct tags when passed props', () => {
  return nasaApi.getApod().then(response => {
    const wrapper = mount(
      <MemoryRouter>
        <AstronomyPicOfTheDay posts={response}/>
      </MemoryRouter>
    )
    expect(wrapper.containsAllMatchingElements([
      <div key={'title'}>
        <h1></h1>
        <img/>
        <h3></h3>
        <h2></h2>
      </div>
    ]))
  })
})

test('AstronomyPicOfTheDay renders empty div when no props recieved', () => {
  const wrapper = shallow(
    <MemoryRouter>
      <AstronomyPicOfTheDay/>
    </MemoryRouter>)
  expect(wrapper.containsAllMatchingElements([
    <div></div>
  ]))
})
