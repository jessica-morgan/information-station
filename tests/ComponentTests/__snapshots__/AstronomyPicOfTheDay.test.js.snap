// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstronomyPicOfTheDay renders correctly 1`] = `
<div>
  <div>
    <img
      className="loading-indicator"
      src="/loading.gif"
    />
  </div>
</div>
`;
